app/test/privacy_policy/table_of_contents/toc_section_expansion_test.dart

```diff
@@ -0,0 +1,593 @@
+              _SectionResult('Foo', isExpanded: true),
+            ],
+          );
+        });
+      });
+
+      test(
+          'When manually toggling the expansion state on a manually expanded section (currently not read) it collapses',
+          () {
+        forEveryExpansionBehavior(() {
+          final sections = [
+            _Section(
+              'Quz',
+              subsections: const [
+                _Section('Xyzzy'),
+              ],
+            )
+          ];
+          tocController.build(sections);
+
+          tocController.toggleExpansionOfSection('Quz');
+          tocController.toggleExpansionOfSection('Quz');
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Quz', isExpanded: false),
+            ],
+          );
+        });
+      });
+      test(
+          'When manually expanding a section it stays open even if it the user finishes reading it',
+          () {
+        forExpansionBehavior(
+            [ExpansionBehavior.leaveManuallyOpenedSectionsOpen], () {
+          final sections = [
+            _Section(
+              'Foo',
+              subsections: const [
+                _Section('Bar'),
+                _Section('Baz'),
+              ],
+            ),
+          ];
+          tocController.build(sections);
+
+          tocController.toggleExpansionOfSection('Foo');
+          tocController.markAsCurrentlyRead('Foo');
+          tocController.markAsCurrentlyRead(null);
+
+          expect(
+            tocController.currentState.sections,
+            [
+              _SectionResult('Foo', isExpanded: true),
+            ],
+          );
+        });
+      });
+      test(
+        'When manually collapsing a section that is currently read it will stay closed when switching between its subchapters',
+        () {
+          forExpansionBehavior(
+              [ExpansionBehavior.leaveManuallyOpenedSectionsOpen], () {
+            final sections = [
+              _Section(
+                'Foo',
+                subsections: const [
+                  _Section('Bar'),
+                  _Section('Baz'),
+                ],
+              ),
+            ];
+            tocController.build(sections);
+
+            // Will expand it automatically
+            tocController.markAsCurrentlyRead('Foo');
+            // We collapse it manually
+            tocController.toggleExpansionOfSection('Foo');
+
+            // We "scroll" around in the subchapters and the main chapter of Foo
+            tocController.markAsCurrentlyRead('Bar');
+            tocController.markAsCurrentlyRead('Baz');
+            tocController.markAsCurrentlyRead('Foo');
+            tocController.markAsCurrentlyRead('Bar');
+
+            expect(
+              tocController.currentState.sections,
+              [
+                _SectionResult('Foo', isExpanded: false),
+              ],
+            );
+          });
+        },
+      );
```

Comment:
```suggestion
      );

```

Add new line after test.